### PR TITLE
[gcs] Stop running duplicate tests

### PR DESCRIPTION
**Pull request by fishbone**
## Why are these changes needed?
After the flags are turned on automatically, stop the old testing.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: .buildkite/pipeline.yml [hunk 1]
```diff
@@ -322,71 +322,6 @@
     - docker run --rm --privileged -v /ray/containers:/var/lib/containers -v /ray:/ray --entrypoint /bin/bash
       rayproject/ray-worker-container:nightly-py36-cpu /ray/ci/travis/test-worker-in-container.sh
 
-- label: ":redis: HA GCS (Dashboard + Serve Tests)"
```
Comment (mwtian) [inline .buildkite/pipeline.yml:325]: Maybe in a separate PR, can we leave some tests running without GCS-Ray? At least until we are ready to delete the deprecated codepath.

**Review by mwtian — APPROVED** [review]

**Review by rkooo567 — APPROVED** [review]
I am so happy for this change, and congratulations @mwtian @iycheng for making this work!

File: .buildkite/pipeline.yml (hunk 1 above)
Comment (mwtian) [reply .buildkite/pipeline.yml:325]: Sounds good.

**Comment by rkooo567** [comment]
There are many failing tests.

**Comment by iycheng** [comment]
:( I'm not sure what's happening there. Let me take a look.

**Comment by iycheng** [comment]
It seems the old pubsub went wrong. I'm updating the test only cover gcs_storage=redis/memory because we are not going to maintain the old code path.

**Comment by iycheng** [comment]
It seems like something is not working here after we turnining on the flags. :( I'll split the PR and 1) stop ha nightly tests; 2) fix the failures.

**Comment by iycheng** [comment]
https://github.com/ray-project/ray/pull/22636

**Pull request closed without merging**